File: js/task3c/S3_PIPE_03.js
function trim(text) {
  return text.trim();
}

function lowercase(text) {
  return text.toLowerCase();
}

function normalizeSpaces(text) {
  return text.replace(/\s+/g, ' ');
}

function processText(text) {
  return normalizeSpaces(lowercase(trim(text)));
}
const inputText = "   Hello   WORLD!   This   is  a TEST.  ";
const result = processText(inputText);
console.log(result); 

File: python/task3c/S3_PIPE_03.py
def trim(text):
  return text.strip()

def lowercase(text):
  return text.lower()

def normalize_spaces(text):
  import re
  return re.sub(r"\s+", " ", text)

def process_text(text):
  return normalize_spaces(lowercase(trim(text)))

input_text = "   Hello   WORLD!   This   is  a TEST.  "
result = process_text(input_text)
print(result)
